itl/plugins-contrib.d/systemd.conf

```diff
@@ -1,51 +1,99 @@
+		}
 		"--unit" = {
 			value = "$systemd_unit$"
 			description = "Name of the systemd unit that is being tested."
 		}
+		"--include-type" = {
+			value = "$systemd_include_type$"
+			description = "Unit types to be tested (for example: `service`, `timer`). This option can be applied multiple times."
+			repeat_key = true
+		}
 		"--exclude" = {
 			value = "$systemd_exclude_unit$"
-			description = "Exclude a systemd unit from the checks. This option can be applied multiple times. Also supports regular expressions."
+			description = "Exclude a systemd unit from the checks, regular expressions are supported. This option can be applied multiple times."
 			repeat_key = true
 		}
-		"--no-startup-time" = {
-			set_if = "$systemd_no_startup_time$"
-			description = "Don’t check the startup time. Using this option the options `systemd_warning` and `systemd_critical` have no effect. (Default: `false`)"
+		"--exclude-unit" = {
+			value = "$systemd_exclude_unit_name$"
+			description = "Exclude a systemd unit from the checks. This option can be applied multiple times."
+			repeat_key = true
 		}
-		"--warning" = {
-			value = "$systemd_warning$"
-			description = "Startup time in seconds to result in a warning status. (Default: `60s`)"
+		"--exclude-type" = {
+			value = "$systemd_exclude_type$"
+			description = "Exclude a systemd unit type (for example: `service`, `timer`)"
 		}
-		"--critical" = {
-			value = "$systemd_critical$"
-			description = "Startup time in seconds to result in a critical status. (Default: `120s`)"
+		"--state" = {
+			value = "$systemd_state$"
+			description = "Specify the active state that the systemd unit must have (for example: `active`, `inactive`)"
 		}
+
+		/* Timers related options */
 		"--dead-timers" = {
-			set_if = "$systemd_dead_timers$"
-			description = "Detect dead / inactive timers. (Default: `false`)"
+			value = "$systemd_dead_timers$"
+			description = "Detect dead / inactive timers, see `systemd_dead_timers_{warning,critical}`. (Default `false`)"
 		}
-		"--dead-timers-warning" = {
+		"--timers-warning" = {
 			value = "$systemd_dead_timers_warning$"
-			description = "Time ago in seconds for dead / inactive timers to trigger a warning state (by default 6 days)."
+			description = "Time ago in seconds for dead / inactive timers to trigger a warning state. (Default 6 days)"
 		}
-		"--dead-timers-critical" = {
+		"--timers-critical" = {
 			value = "$systemd_dead_timers_critical$"
-			description = "Time ago in seconds for dead / inactive timers to trigger a critical state (by default 7 days)."
+			description = "Time ago in seconds for dead / inactive timers to trigger a critical state. (Default 7 days)"
 		}
-		"-v" = {
-			set_if = {{ macro("$systemd_verbose_level$") == 1 }}
-			description = "Increase verbosity level (Accepted values: `1`, `2` or `3`). Defaults to none."
+
+		/* Startup time related options */
+		"--no-startup-time" = {
+			value = "$systemd_no_startup_time$"
```

Comment:
I'm not sure if this change was made accidentally due to the moving around, but `--no-startup-time` is a boolean type so this must be `set_if = ...` as it was before and not `value`, so that Icinga 2 simply maps it to `check_systemd ... --no-startup-time` if it's set to a value that evaluates to true, instead of `check_systemd ... --no-startup-time 'WHATEVER_VALUE_IT_IS_SET_TO'`.

```bash
[2024-09-06 07:57:55 +0000] notice/Process: PID 1335540 ('/usr/local/sbin/check_systemd' '--no-startup-time' 'true') terminated with exit code 128
```